### ISSUE TITLE
What's holding the Dart 3.2 release back?

### ISSUE THREAD

**Issue by passsy** [dart-lang/sdk#54012]
**tl;dr** Dart 3.2 is ready but not released. Flutter is blocking Dart developers for no reason.

According to the `CHANGELOG.md`, Dart 3.2 is ready for release. All development is focused on the next version Dart 3.3.
Since October, there have been only minor changes on top of version `3.2.0-210.1.beta`

Version | OS | Arch | Date
-- | -- | -- | --
3.2.0-210.1.beta (ref 5c23602) | macOS | x64 | Oct 3, 2023
3.2.0-210.2.beta (ref 620c172) | macOS | x64 | Oct 11, 2023
3.2.0-210.3.beta (ref 668bef3) | macOS | x64 | Oct 18, 2023
3.2.0-210.4.beta (ref a30d2ed) | macOS | x64 | Nov 1, 2023

However, I cannot use the Dart 3.2 normally, because Dart 3.2 has not been released officially. This causes multiple annoyances:

- As Dart fan, I want to use the latest Dart version and give early feedback from production software. But development is far ahead and hot-fixed may have to be cherry-picked into multiple Dart versions.
- As Dart developer, who builds CLIs, backends and websites, I have to use the Dart beta docker container, which is not ideal for production environments.
- As package author, I'd like to build and release packages with new language features. I don't want to flag the releases as pre-releases as suggested by `pub publish` for the time between code freeze and actually announcing it.

--- 

I have the feeling that Dart release announcements are postponed to sync with Flutter releases. That might be a good decision from a marketing perspective. But as Dart developer, who's not using the Flutter SDK in all projects, it is a bit frustrating.

Please bring back continuous Dart releases every 2-3 months, independent from Flutter, like back in the Dart 1.X days.

**Comment by mit-mit** [comment]
Hi Pascal,

> Dart 3.2 is ready but not released. Flutter is blocking Dart developers for no reason.

This is not true. While the release is indeed getting very close to being ready -- hence no changes to it -- there is still final testing being completed.

> According to the CHANGELOG.md, Dart 3.2 is ready for release

The changelog doesn't tell you when it's "ready" from a quality point of view

>  All development is focused on the next version Dart 3.3

Yes, the main channel is focused on 3.3. But we still do cherry picks from main to the release channels.

> As Dart fan, I want to use the latest Dart version and give early feedback from production software.

You can use beta for this. As you say yourself it's very, very close (potentially identical) to next stable, given we're on the last beta of the next stable.

> I have the feeling that Dart release announcements are postponed to sync with Flutter releases

It is correct that we for the past several years have sym-shipped Dart and Flutter SDKs on the same date. But this is not some "marketing" tactic, it's because a significant amount of Dart quality signal these days is via Flutter testing.

> Please bring back continuous Dart releases every 2-3 months

We ship roughly every 3 months, and have done so for quite a while (the long gap after 2.19 was to wrap breaking changes in 3.0):

- Dart 3.1: 6 August 2023
- Dart 3.0: 10 May 2023
- Dart 2.19: 25 January 2023 
- Dart 2.18: 30 August 2022
- Dart 2.17: 11 May 2022
- Dart 2.16: 3 February 2022

**Comment by mit-mit** [comment]
One thought: would more transparency over the release cadence help address this?

**Comment by mit-mit** [comment]
Generally *any* beta release is worth testing. 

The general release process is that we cut a new beta release about once a month -- on occasion more frequently if we need to cherry-pick fixes into beta. And then in the end of a release cycle, the last beta channel release is promoted to be the stable release.

I think to you points -- and in extension of my comment above -- what we're missing is transparency around / comms about, when we've cut the intended last beta release, and are in the final holding period with no known issues and just final release validation missing? This would be somewhat similar to how we announced Release Candidate builds back in 2016-2017. Would that be helpful?

**Comment by mnordine** [comment]
Just chiming in here, yes, that would be helpful, 👍

**Comment by parlough** [comment]
I will say that Chrome/Chromium's status page is really useful with beta and stable timelines: https://chromestatus.com/roadmap. Something similar would be nice even if the exact dates weren't as precise. It helps developers quickly understand when and what to test, what deprecations to prepare for, or what breaking changes to be aware of.

**Comment by passsy** [comment]
The Release Candidate announcements of Dart 1 were fantastic 👍

[![Screen-Shot-2023-11-13-17-08-37 90](https://github.com/dart-lang/sdk/assets/1096485/76e94150-4fc7-4692-8cc1-416396e2a186)](https://groups.google.com/a/dartlang.org/g/announce/c/FNIvnC7FR3k)

**Comment by h3x4d3c1m4l** [comment]
Looks like it has just been released!

**Comment by FMorschel** [comment]
> One thought: would more transparency over the release cadence help address this?

I believe so. I even created an [issue yesterday](https://github.com/dart-lang/site-www/issues/5370) about that.
